src/pybind/mgr/cephadm/module.py

```diff
@@ -1715,6 +1716,16 @@ def run_cmd(cmd_args: dict) -> None:
         # if host was in offline host list, we should remove it now.
         self.offline_hosts_remove(host)
         self.event.set()  # refresh stray health check
+        if rm_crush_entry:
+            try:
+                self.check_mon_command({
+                    'prefix': 'osd crush remove',
+                    'name': host,
+                })
+            except MonCommandFailed as e:
+                self.log.error(f'Couldn\'t remove host {host} from CRUSH map: {str(e)}')
+                return (f'Host {host} was removed from cephadm control '
+                        f'but cephadm failed to remove it from the CRUSH map: {str(e)}')
```

Comment:
+1 Seems to me that it would be more repeatable that way. In the case where the host is removed but not removed from crush it may be more work / confusing to have to deal with the fact that the host is gone but there's still a crush entry lying around.

Reply:
Okay, I've since moved it to happen before we remove the host from the inventory, and adjusted the return message and docs section accordingly.